lib/services/identity_service.dart

```diff
@@ -37,6 +41,50 @@ class IdentityService with LogMixin, ChangeCallbackProvider {
     return updatedName;
   }
 
+  Future<String> get publicKeyPem async =>
+      await _repository.get<String>(publicKeyKey) ?? '';
+
+  Future<RSAPublicKey?> get publicKey async {
+    if (await publicKeyPem == '') {
+      return null;
+    }
+
+    try {
+      return KeyHelper().decodePublicKeyFromPem(await publicKeyPem);
+    } on Exception catch (error) {
+      logger.severe('could not parse privateKey $error');
+    }
```

Comment:
Should store the `await publicKeyPem` in a variable that is doesn't have to be done twice.